components/form/Details.tsx

```diff
@@ -0,0 +1,197 @@
+                step: 'any',
+                required: true,
+                onKeyDown: preventMinus,
+                pattern: '[0-9]*',
+                inputMode: 'numeric',
+              }}
+            />
+
+            <Input
+              id="solar-panel-tilt"
+              label="Solar panel tilt"
+              description="(Degrees above the horizontal)"
+              value={String(panelDetails.tilt)}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                setFormData({ ...panelDetails, tilt: e.currentTarget.value })
+              }
+              inputProps={{
+                type: 'number',
+                placeholder: '40º',
+                min: '0',
+                max: '360',
+                step: 'any',
+                required: true,
+                onKeyDown: preventMinus,
+                pattern: '[0-9]*',
+                inputMode: 'numeric',
+              }}
+            />
+
+            <Input
+              label={
+                <>
+                  Solar panel capacity
+                  <span className="text-xs font-normal"> (optional)</span>
+                </>
+              }
+              id="solar-panel-capacity"
+              value={String(panelDetails.capacity)}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                setFormData({
+                  ...panelDetails,
+                  capacity: e.currentTarget.value,
+                })
+              }
+              inputProps={{
+                type: 'number',
+                placeholder: '2800 kW',
+                min: '0',
+                step: 'any',
+                onKeyDown: preventMinus,
+                pattern: '[0-9]*',
+                inputMode: 'numeric',
+              }}
+            />
+            <button
+              disabled={didSubmit}
+              className="md:hidden mt-8 font-bold text-xl w-full peer-invalid:bg-ocf-gray-300 transition duration-150 bg-ocf-yellow dark:disabled:bg-ocf-gray-300 dark:bg-ocf-yellow shadow h-14 border-gray-200 focus:ring-4 focus:outline-none focus:ring-gray-100 rounded-md px-5 py-2.5 text-center inline-flex items-center justify-center mr-2 mb-2"
+            >
+              {didSubmit && <Spinner width={5} height={5} margin={4} />}
+              Finish
+              {didSubmit && <div className="w-5 mx-4" />}
+            </button>
+            <div className="md:hidden h-20 w-full" />
+            <Modal show={showModal} setShow={setShowModal} />
+            <div className="hidden absolute md:flex md:flex-row md:justify-between bottom-12 left-1/2 -translate-x-1/2 w-3/4 h-14">
+              <Button disabled={false} onClick={lastPageCallback}>
+                Back
+              </Button>
+              <Button disabled={didSubmit}>
+                {didSubmit && <Spinner width={5} height={5} margin={2} />}
+                Finish
+                {didSubmit && <div className="w-5 mx-2" />}
+              </Button>
```

Comment:
Similar clipping on these buttons to on the Location page (with fixing steps mentioned below):
<img width="235" alt="image" src="https://user-images.githubusercontent.com/23221268/227702890-60d00d7b-2c89-4ec0-8abf-c7478c71cd77.png">

IMO remove the position `absolute` on the div above this, that should do most of the work. I just did some deeper looking tho and honestly I'd move this button row outside the form. That way, you can put it at the bottom of the page and give it full width and appropriately justify-between the buttons. You can tie the submit button here back to the form by giving the form an `id` and using the [form attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#:~:text=The%20%3Cform%3E%20element%20to%20associate%20the%20button%20with%20(its%20form%20owner).%20The%20value%20of%20this%20attribute%20must%20be%20the%20id%20of%20a%20%3Cform%3E%20in%20the%20same%20document.%20(If%20this%20attribute%20is%20not%20set%2C%20the%20%3Cbutton%3E%20is%20associated%20with%20its%20ancestor%20%3Cform%3E%20element%2C%20if%20any.))